src/code/z_camera.c

```diff
@@ -1216,212 +1301,226 @@ f32 func_800CD6CC(Actor* actor) {
-    sp50.z = Math_CosS(temp_s1->rot.y) * arg3;
-    sp50.y = trackHeight + yOffset;
-    Camera_LERPCeilVec3f(&sp50, &camera->atActorOffset, camera->xzOffsetUpdateRate, camera->yOffsetUpdateRate, 0.1f);
+    Camera_ScaledStepToCeilVec3f(&focalActorAtOffsetTarget, &camera->focalActorAtOffset, camera->xzOffsetUpdateRate,
+                                 camera->yOffsetUpdateRate, 0.1f);
 
-    sp44.x = temp_s1->pos.x + camera->atActorOffset.x;
-    sp44.y = temp_s1->pos.y + camera->atActorOffset.y;
-    sp44.z = temp_s1->pos.z + camera->atActorOffset.z;
-    sp38.x = temp_s1->pos.x;
-    sp38.y = sp44.y;
-    sp38.z = temp_s1->pos.z;
+    atTarget.x = focalActorPosRot->pos.x + camera->focalActorAtOffset.x;
+    atTarget.y = focalActorPosRot->pos.y + camera->focalActorAtOffset.y;
+    atTarget.z = focalActorPosRot->pos.z + camera->focalActorAtOffset.z;
 
-    if (Camera_BGCheck(camera, &sp38, &sp44)) {
-        sp44.x -= camera->atActorOffset.x - (sp44.x - sp38.x);
-        sp44.z -= camera->atActorOffset.z - (sp44.z - sp38.z);
+    collisionFromPos.x = focalActorPosRot->pos.x;
+    collisionFromPos.y = atTarget.y;
+    collisionFromPos.z = focalActorPosRot->pos.z;
+
+    if (Camera_BgCheck(camera, &collisionFromPos, &atTarget)) {
+        atTarget.x -= camera->focalActorAtOffset.x - (atTarget.x - collisionFromPos.x);
+        atTarget.z -= camera->focalActorAtOffset.z - (atTarget.z - collisionFromPos.z);
     }
 
-    atLERPStepScale = camera->atLERPStepScale;
-    Camera_LERPCeilVec3f(&sp44, &camera->at, atLERPStepScale, atLERPStepScale, 0.2f);
-    return true;
+    Camera_ScaledStepToCeilVec3f(&atTarget, &camera->at, camera->atLerpStepScale, camera->atLerpStepScale, 0.2f);
+
+    return 1;
 }
 
-s32 func_800CDB6C(Camera* camera, VecSph* arg1, f32 yOffset, f32 arg3, f32* arg4, s16 arg5) {
-    f32 new_var;
-    Vec3f sp70;
-    Vec3f sp64;
-    f32 temp_f0;
-    f32 temp_f12;
-    f32 sp58;
-    f32 sp54;
-    PosRot* trackActorPosRot = &camera->trackActorPosRot;
-    f32 trackHeight = Camera_GetTrackedActorHeight(camera);
-    VecSph sp44;
+/**
+ * Adjusts the camera's at position for Camera_Parallel1
+ */
+s32 Camera_CalcAtForParallel(Camera* camera, VecSph* arg1, f32 yOffset, f32 xzOffsetMax, f32* focalActorPosY,
+                             s16 flags) {
+    f32 pad;
+    Vec3f focalActorAtOffsetTarget;
+    Vec3f atTarget;
+    f32 fovHeight;
     Vec3f* at = &camera->at;
-
-    if (arg5 & 0x40) {
-        sp44.r = func_800CCCEC(camera, arg5 & 0x10);
-        sp44.yaw = trackActorPosRot->rot.y + 0x4000;
-        sp44.pitch = 0;
-        OLib_VecSphGeoToVec3f(&sp70, &sp44);
+    f32 deltaY;
+    f32 eyeAtDistXZ;
+    PosRot* focalActorPosRot = &camera->focalActorPosRot;
+    f32 focalActorHeight = Camera_GetFocalActorHeight(camera);
+    VecSph focalActorAtOffsetTargetSph;
+
+    // Calculate the `focalActorAtOffsetTarget`
+    if (flags & 0x40) {
+        focalActorAtOffsetTargetSph.r = func_800CCCEC(camera, flags & 0x10);
```

Comment:
I interpreted the design of these `CalcAt` functions in a way that in general the flags can be used by different cam-update functions. In OoT for example, `Camera_CalcAtForLockOn` is used by both `BATTLE1_FLAG` and `KEEPON1_FLAG` which need to be synced to align with what the function `Camera_CalcAtForLockOn` does with those flags.

That's why I avoided uses `PARALLEL1_FLAG_` flags here since it is outside the Parallel1 function, but yes in theory because this function is only used by Parallel1, that flag could be used here. I think I would wait until the `CalcAt` function flags are better documented as a whole to see if I want to use the PARALLEL1 flag here if that makes sense?

Reply:
Ohhh, I see... Why does camera has to be so damn cursed?
I think it is fine to leave the raw value for now then. Maybe you could add a comment explaining it a bit